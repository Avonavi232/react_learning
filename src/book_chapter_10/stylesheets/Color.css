section.color{-webkit-box-shadow:3px 3px 8px 1px #b2b2b2;box-shadow:3px 3px 8px 1px #b2b2b2;height:235px;border:2px solid #e5e5e5;border-bottom-color:#7f7f7f;border-right-color:#7f7f7f;margin:1em;padding:0;-ms-flex-preferred-size:100%;flex-basis:100%;display:-webkit-box;display:-ms-flexbox;display:flex;-ms-flex-wrap:wrap;flex-wrap:wrap;position:relative}@media screen and (min-width: 700px){section.color{-ms-flex-preferred-size:calc(50% - 2em - 4px);flex-basis:calc(50% - 2em - 4px)}}@media screen and (min-width: 1100px){section.color{-ms-flex-preferred-size:calc(33.33333% - 2em - 4px);flex-basis:calc(33.33333% - 2em - 4px)}}@media screen and (min-width: 1600px){section.color{-ms-flex-preferred-size:calc(25% - 2em - 4px);flex-basis:calc(25% - 2em - 4px)}}@media screen and (min-width: 2000px){section.color{-ms-flex-preferred-size:calc(16.66666% - 2em - 4px);flex-basis:calc(16.66666% - 2em - 4px)}}section.color h1{font-family:Verdana, sans-serif;text-align:center;padding:.5em;-ms-flex-preferred-size:calc(100% - 1em);flex-basis:calc(100% - 1em)}section.color button{position:absolute;top:.5em;right:.5em;-ms-flex-preferred-size:calc(12% - 1em);flex-basis:calc(12% - 1em);text-align:right;background:#fff;background:-webkit-gradient(linear, left top, left bottom, from(#fff), to(#e5e5e5));background:linear-gradient(to bottom, #fff 0%, #e5e5e5 100%);padding:.75em;border-radius:.5px}section.color>div:last-child{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;width:100%;margin-top:.5em}section.color div.color{height:100px;-ms-flex-preferred-size:100%;flex-basis:100%}section.color div.time-ago{position:absolute;bottom:.5em;right:.5em;font-family:monospace;color:#7f7f7f}

/*# sourceMappingURL=Color.css.map */