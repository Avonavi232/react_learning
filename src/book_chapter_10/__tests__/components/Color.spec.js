import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import {compose} from 'redux';

import Color from '../../components/ui/Color';


describe('<Color/> Component UI', function () {
    const shallowExpect = compose(expect, toJSON, shallow);

    it('should render correct properties', function () {
        shallowExpect(
            <Color
                title="Test Color"
                color="#ffffff"
                rating={3}
                timestamp={new Date().toDateString()}
            />
        ).toMatchSnapshot();
    });
});