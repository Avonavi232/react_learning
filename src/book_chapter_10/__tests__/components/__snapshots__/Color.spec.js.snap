// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color/> Component UI should render correct properties 1`] = `
<section
  className="color"
>
  <h1>
    Test Color
  </h1>
  <button
    onClick={[Function]}
  >
    <FaTrashO />
  </button>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  />
  <TimeAgo
    timestamp="Fri Jun 22 2018"
  />
  <div>
    <StarRating
      onRate={[Function]}
      starsSelected={3}
    />
  </div>
</section>
`;
